modalc-lib/modal-vector.rkt

```diff
@@ -45,10 +43,54 @@
                                     ((vector-ref inner-ctc-proj/blame index) value neg-party)]
                                    [else value]))
                            (λ (vec index value)
-                             (cond [(should-apply-ctc?/set vec)
+                             (cond [(should-apply-ctc?/set (list index value))
                                     ((vector-ref inner-ctc-proj/blame index) value neg-party)]
                                    [else value])))))))
 
+(define (modal-vectorof should-apply-ctc? . inner-ctc)
+  (apply modal-vectorof* inner-ctc #:ref/set should-apply-ctc?))
+
+(define (modal-vectorof* #:ref [ref-ctc? #f]
+                        #:set [set-ctc? #f]
+                        #:ref/set [all-ctc? #f]
+                        . inner-ctc)
```

Comment:
`vectorof` only accepts a single inner contract. The relationship between `vectorof` and `vector/c` is like that between `listof` and `list/c`